### PR TITLE
Contest-specific DCP to FPGA Interchange Format Utility

### PR DESCRIPTION
**Pull request by eddieh-xlnx**
This utility is used to convert a fully-routed Vivado Design Checkpoint (DCP) into the FPGA Interchange Format's Logical and Physical Netlists compatible with the FPGA24 contest.

Specifically, in line with the [competition format](https://xilinx.github.io/fpga24_routing_contest/details.html#pre-routing) the Physical Netlist will only contain the inter-site routing (i.e. PIPs) for global and static nets. All other nets will have their inter-site routing removed. Intra-site routing (not within the scope of this contest) is left as-is.

Example usage:
```
./gradlew -Dmain=com.xilinx.fpga24_routing_contest.DcpToFPGAIF :run --args="boom_soc_routed.dcp boom_soc.netlist boom_soc_unrouted.phys"
```

Compared to the benchmarks released in [v1.0](https://github.com/Xilinx/fpga24_routing_contest/releases/tag/v1.0), the benchmarks in the upcoming release contain nets with alternate sources set up, where available. Alternate sources typically refer to the ability for a LUT to exit its SLICE using up to 2 pins -- for an `A6LUT` these pins are `A_O` (always available) and `AMUX` (available if not used for another purpose). Considering both pins, where available, can improve routability, though there are no restrictions on which of the two (or whether both) source pins are used to connect to which sink pins.

Use of this utility is not necessary to enter the contest, though contestants may choose to use Vivado to compile additional benchmarks beyond those that we supply in order to increase test coverage/acquire more training data, etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,13 @@ Please see website at https://xilinx.github.io/fpga24_routing_contest.
...
-All content in this repository except for third-party software (Gradle) is licensed under:
+---
+All content in this repository except for third-party software ([Gradle](https://gradle.org)) is licensed under:
 ```
 SPDX-License-Identifier: MIT
 ```
+---
+
+Utilities provided:
+* [`net_printer`](https://github.com/Xilinx/fpga24_routing_contest/tree/dcp_to_fpgaif/net_printer) -- inspect the routing of nets in a Physical Netlist.
```
Comment (zakn-amd) [inline README.md:15]: Is there a reason that this links to the `dcp_to_fpgaif` branch instead of `master`?

Comment (eddieh-xlnx) [reply README.md:15]: Good catch! Will fix it now.

Comment (eddieh-xlnx) [inline README.md:15]:
```suggestion
* [`net_printer`](https://github.com/Xilinx/fpga24_routing_contest/tree/master/net_printer) -- inspect the routing of nets in a Physical Netlist.
```

File: README.md [hunk 2]
```diff
@@ -4,7 +4,13 @@ Please see website at https://xilinx.github.io/fpga24_routing_contest.
...
 
-All content in this repository except for third-party software (Gradle) is licensed under:
+---
+All content in this repository except for third-party software ([Gradle](https://gradle.org)) is licensed under:
 ```
 SPDX-License-Identifier: MIT
 ```
+---
+
+Utilities provided:
```
Comment (eddieh-xlnx) [inline README.md:14]:
```suggestion
Utilities:
```

**Pull request merged**